trunk/SUAVE/Components/Energy/Networks/Vectored_Thrust.py

```diff
@@ -0,0 +1,326 @@
+        segment.state.conditions.propulsion.propeller_power_coefficient = segment.state.unknowns.propeller_power_coefficient
+        segment.state.conditions.propulsion.battery_voltage_under_load  = segment.state.unknowns.battery_voltage_under_load
+        segment.state.conditions.propulsion.throttle                    = segment.state.unknowns.throttle  
+ 
+        return
+      
+    def unpack_unknowns(self,segment):
+        """ This is an extra set of unknowns which are unpacked from the mission solver and send to the network.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.unknowns.propeller_power_coefficient [None]
+            state.unknowns.battery_voltage_under_load  [volts]
+    
+            Outputs:
+            state.conditions.propulsion.propeller_power_coefficient [None]
+            state.conditions.propulsion.battery_voltage_under_load  [volts]
+    
+            Properties Used:
+            N/A
+        """                  
+
+        # Here we are going to unpack the unknowns (Cp) provided for this network
+        segment.state.conditions.propulsion.propeller_power_coefficient = segment.state.unknowns.propeller_power_coefficient
+        segment.state.conditions.propulsion.battery_voltage_under_load  = segment.state.unknowns.battery_voltage_under_load
+        segment.state.conditions.propulsion.throttle                    = segment.state.unknowns.throttle
+          
+        
+        return
+    
+    def residuals(self,segment):
+        """ This packs the residuals to be send to the mission solver.
+    
+            Assumptions:
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state.conditions.propulsion:
+                motor_torque                          [N-m]
+                propeller_torque                      [N-m]
+                voltage_under_load                    [volts]
+            state.unknowns.battery_voltage_under_load [volts]
+            
+            Outputs:
+            None
+    
+            Properties Used:
+            self.voltage                              [volts]
+        """        
+        
+        # Here we are going to pack the residuals (torque,voltage) from the network
+        
+        # Unpack
+        q_motor   = segment.state.conditions.propulsion.motor_torque
+        q_prop    = segment.state.conditions.propulsion.propeller_torque
+        v_actual  = segment.state.conditions.propulsion.voltage_under_load
+        v_predict = segment.state.unknowns.battery_voltage_under_load
+        v_max     = self.voltage
+        
+        # Return the residuals
+        segment.state.residuals.network[:,0] = q_motor[:,0] - q_prop[:,0]
+        segment.state.residuals.network[:,1] = (v_predict[:,0] - v_actual[:,0])/v_max
+        
+        #print (segment.state.residuals.network[:,0])
```

Comment:
Extra comments

Reply:
Comment removed and functions cleaned up